src/main/kotlin/no/nav/syfo/infrastructure/database/repository/AktivitetskravRepository.kt

```diff
@@ -249,3 +285,83 @@ private fun ResultSet.toPAktivitetskrav(): PAktivitetskrav = PAktivitetskrav(
+        createdAt = getObject("created_at", OffsetDateTime::class.java),
+        createdBy = getString("created_by"),
+        status = status,
+        beskrivelse = getString("beskrivelse"),
+        arsaker = getString("arsaker").split(",").map(String::trim).filter(String::isNotEmpty)
+            .map { it.toVurderingArsak(status) },
+        frist = getDate("frist")?.toLocalDate(),
+        varsel = null,
+    )
+}
+
+fun ResultSet.toAktivitetskravWithVurderinger(): List<PAktivitetskrav> {
+    val aktivitetskrav = mutableMapOf<Int, PAktivitetskrav>()
+    while (this.next()) {
+        val varselUuid = getString("varsel_uuid")
+        val varsel = varselUuid?.let {
+            PAktivitetskravVarsel(
+                id = getInt("varsel_id"),
+                uuid = UUID.fromString(getString("varsel_uuid")),
+                createdAt = getObject("varsel_created_at", OffsetDateTime::class.java),
+                updatedAt = getObject("varsel_updated_at", OffsetDateTime::class.java),
+                aktivitetskravVurderingId = getInt("varsel_aktivitetskrav_vurdering_id"),
+                journalpostId = getString("varsel_journalpost_id"),
+                document = configuredJacksonMapper().readValue(
+                    getString("varsel_document"),
+                    object : TypeReference<List<DocumentComponentDTO>>() {}
+                ),
+                publishedAt = getObject("varsel_published_at", OffsetDateTime::class.java),
+                svarfrist = getDate("varsel_svarfrist")?.toLocalDate(),
+                expiredVarselPublishedAt = getObject("varsel_expired_varsel_published_at", OffsetDateTime::class.java),
+                type = getString("varsel_type"),
+            )
+        }
+        val vurderingUuid = getString("vurdering_uuid")
+        val vurdering =
+            vurderingUuid?.let {
+                val vurderingStatus = AktivitetskravStatus.valueOf(getString("vurdering_status"))
+                PAktivitetskravVurdering(
+                    id = getInt("vurdering_id"),
+                    uuid = UUID.fromString(getString("vurdering_uuid")),
+                    aktivitetskravId = getInt("vurdering_aktivitetskrav_id"),
+                    createdAt = getObject("vurdering_created_at", OffsetDateTime::class.java),
+                    createdBy = getString("vurdering_created_by"),
+                    status = AktivitetskravStatus.valueOf(getString("vurdering_status")),
+                    beskrivelse = getString("vurdering_beskrivelse"),
+                    arsaker = getString("vurdering_arsaker").split(",").map(String::trim).filter(String::isNotEmpty)
+                        .map { it.toVurderingArsak(vurderingStatus) },
+                    frist = getDate("vurdering_frist")?.toLocalDate(),
+                    varsel = varsel,
+                )
+            }
+
+        val aktivitetskravId = getInt("aktivitetskrav_id")
+        aktivitetskrav[aktivitetskravId]?.let {
```

Comment:
Litt vanskelig å lese/henge med på denne. Kanskje man kunne dratt ut noen variabler for å forbedre eller om man kunne noe `putIf*` / `computeIf`*-metoder på map i stedet.

Reply:
Lurt! Enig i at denne ble litt kryptisk 🙈